### PR TITLE
Realm Web: MongoDB & HTTP services

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This PR is adding remote MongoDB and HTTP services to Realm Web, closing more of #2699.
Please note: This has already been released internally as Realm Web 0.1.0 and for the tasks not checked below, I suggest we postpone this to a separate PR, once the API has stabilized.

## ☑️ ToDos
* [ ] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Comment by kraenhansen** [comment]
> looks alright. The code formatting is highlighting we should definitely increase the line length either in this or in a later PR.
> A notable example is this one
> 
> ```
> type NewDocument<T extends Document> = Realm.Services.RemoteMongoDB.NewDocument<
>     T
> >;
> ```

I agree - although you'll most likely be able to find a similar example once it's increased.

**Comment by blagoev** [comment]
or maybe the formatter in this case is missing some settings. I can't say for sure. Since why not split the lines like this
```
type NewDocument<T extends Document> = 
      Realm.Services.RemoteMongoDB.NewDocument<T>;
```

File: packages/realm-web/src/services/RemoteMongoDBService.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    return new RemoteMongoDBCollection<T>(
+        transport,
+        serviceName,
+        databaseName,
+        collectionName,
+    );
+}
+
+/**
+ * Creates an Remote MongoDB Database
```
Comment (kneth) [inline packages/realm-web/src/services/RemoteMongoDBService.ts:261]:
```suggestion
 * Creates a Remote MongoDB Database
```

File: packages/realm-web/src/services/RemoteMongoDBService.ts [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            null,
+            transport,
+            serviceName,
+            databaseName,
+        ),
+    };
+}
+
+/**
+ * Creates an Remote MongoDB Service
```
Comment (kneth) [inline packages/realm-web/src/services/RemoteMongoDBService.ts:285]:
```suggestion
 * Creates a Remote MongoDB Service
```

File: lib/app.d.ts [hunk 3]
```diff
@@ -124,10 +129,15 @@ declare namespace Realm {
         readonly id: string;
 
         /**
-         * Use this to call functions defined on the MongoDB Realm server.
+         * Use this to call functions within by the MongoDB Realm app.
```
Comment (kraenhansen) [inline lib/app.d.ts:132]:
```suggestion
         * Use this to call functions defined by the MongoDB Realm app.
```

**Pull request merged**